### PR TITLE
Mongo enhancement for sensor data.

### LINKED ISSUE #57: Mongo improvements
* Make a separate schema for sensor data.
* Create endpoints and mongo functions for sensor data.
* Extend `IDevice` type to hold sensor data.
* Add new sensor data object to mongo everytime data is send.

### PR DESCRIPTION
**Pull request by jaanikaukonen**
fixes #57 

## Description
`SensorData` is now pushed in own table in `MongoDB`.

### Backend
* Add new endpoint for fetching sensor data.
    * `/api/devices/sensordata/:id`
* Remove `IDeviceSensorType` since it is no longer needed.
* Refactor `sensorService` to create new object to mongodb when message is received.
* Add `sensorDataSchema`.
* Add `deviceId` in `ISensorData` type.
* Change type of `sensors` to a list of strings in `IDevice` type. 

### Frontend
* Same changes to types as in backend.
* Add `fetchDeviceData` into `ApiHooks.ts`.

### CODE REVIEW

**Review by NikoLindborg — APPROVED** [review]
looks exactly like we discussed

File: back-end/src/schemas/sensorData.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import {Schema, model} from 'mongoose'
+import {ISensorData} from '../types/sensorDataType'
+
+const sensorDataSchema = new Schema<ISensorData>({
```
Comment (aleksikosonen) [inline back-end/src/schemas/sensorData.ts:4]: The `sensorDataSchema` should be renamed to `SensorDataSchema` due to the pipeline being case sensitive.

**Review by aleksikosonen — CHANGES_REQUESTED** [review]
Apart from that one renaming I commented these changes look good.

File: back-end/src/schemas/sensorData.ts (hunk 1 above)
Comment (jaanikaukonen) [reply back-end/src/schemas/sensorData.ts]: Fixed.

**Review by aleksikosonen — APPROVED** [review]

**Pull request merged**